validators/run-a-node/oracle-price-feeder.md

```diff
@@ -1,40 +1,191 @@
 # 🥙 Oracle Price Feeder
 
-As a validator in the active set, you will be required to use the Oracle Price Feeder to submit prices.
+## Introduction
 
-## Installation
+Validators are required to submit price feeds for the on-chain oracle.
+The `price-feeder` app can get price data from multiple providers and
+submit oracle votes to perform this duty.
 
+
+## Requirements
+
+- This guide assumes you are running Ubuntu 22.04.
+- `price-feeder` needs access to a running node's RPC and gRPC ports.
+  This guide assumes you have it on the same machine and uses `localhost`
+  with default ports 26657 and 9090. Change these in `config.toml` as needed.
+- This guide assumes you are configuring the oracle for a mainnet (`kaiyo-1`) validator.
+
+
+## User setup
+
+Best practice is to run node software on an isolated unprivileged user.
+We'll create the `kujioracle` user in this guide; if your username is different 
+change it wherever it appears.
+
+### Create the `kujioracle` user
 ```bash
-git clone https://github.com/Team-Kujira/oracle-price-feeder.git
-cd oracle-price-feeder
-make install
+sudo useradd -m -s /bin/bash kujioracle
 ```
 
-## Configuration
 
-### Price Feeds
+## Build environment setup
 
-Start by copying the default configuration file. This will add price feeds for `ATOM` (the first exchange rate supported by the price-feeder).
+If you've already followed the [Run a node](/validators/run-a-node/run-a-node.md) guide, the only steps 
+you need in this section are **Install go toolchain** #2 and #3. Repeating the 
+others won't hurt if you want to be safe.
 
-This config also contains default values for `gas_price` and `gas_adjustment`
+### Install build packages
+```bash
+sudo apt install -y build-essential git unzip curl
+```
 
+### Install go toolchain
+1. Download and extract go 1.18.5.
+```bash
+curl -fsSL https://golang.org/dl/go1.18.5.linux-amd64.tar.gz | sudo tar -xzC /usr/local
+```
+2. Login as `kujioracle`.
+```bash
+sudo su -l kujioracle
 ```
-cp config.example.toml config.toml
+3. Configure environment variables for `kujioracle`.
+```bash
+cat <<EOF >> ~/.bashrc
+export GOROOT=/usr/local/go
+export GOPATH=\$HOME/go
+export GO111MODULE=on
+export PATH=\$PATH:/usr/local/go/bin:\$HOME/go/bin
+EOF
+source ~/.bashrc
+go version  # should output "go version go1.18.5 linux/amd64"
 ```
 
-### Currency Pairs
 
+## Build `price-feeder`
+
+1. Login as `kujioracle` (skip if you're already logged in).
+```bash
+sudo su -l kujioracle
+```
+2. Build `kujirad` v0.5.0. We'll use the binary to create the keyring file.
```

Comment:
Old version number. Should we structure this so that it's evergreen? There are a lot of places that the version is referenced now, maybe there's a better central source of truth we can link to?

Reply:
Yup definitely. I forked and edited from @mintthemoon wiki (asked him first of course) and that one must have slipped. I updated some to v0.6.4. That said it should be soon changed to v0.7.1. I'll check how to get it done. Will get on it.